### PR TITLE
Benchmark : Modify/Add workflows/actions for github-action-benchmark

### PR DESCRIPTION
**Pull request by varun-sundar-rabindranath**
SUMMARY:
Modify/Add Github actions and workflows required for `github-action-benchmark` UI

TEST PLAN:
Manual testing

### CODE REVIEW

File: .github/workflows/nightly.yml
Comment (varun-sundar-rabindranath) [inline .github/workflows/nightly.yml:16]: This option is to force-push results to the github branch where the benchmark results are maintained for UI

File: .github/actions/nm-github-action-benchmark/action.yml
Comment (varun-sundar-rabindranath) [inline .github/actions/nm-github-action-benchmark/action.yml:52]: This option will kick out values older than 50 entries. i.e. if we only push results on nightly we will only have results for the past 50 days.

File: .github/workflows/nm-benchmark.yml
Comment (varun-sundar-rabindranath) [inline .github/workflows/nm-benchmark.yml:188]: Make sure to switch this to `nm-gh-pages` before landing.

File: .github/workflows/nm-benchmark.yml
Comment (varun-sundar-rabindranath) [inline .github/workflows/nm-benchmark.yml:199]: Make sure to switch this to `nm-gh-pages` before landing.

File: .github/workflows/nm-benchmark.yml
Comment (varun-sundar-rabindranath) [inline .github/workflows/nm-benchmark.yml:164]: Calling out this permission - This is required for the action to be able to write to `nm-gh-pages` branch. However, it grants write access to the entire repo and not just that branch. I couldn't find a finer grained permission.

**Comment by varun-sundar-rabindranath** [comment]
Land this only after https://github.com/neuralmagic/nm-vllm/pull/122 lands.

File: .github/workflows/nm-benchmark.yml
Comment (varun-sundar-rabindranath) [reply .github/workflows/nm-benchmark.yml:164]: @robertgshaw2-neuralmagic

**Review by robertgshaw2-neuralmagic — APPROVED** [review]

File: .github/workflows/nm-benchmark.yml
Comment (varun-sundar-rabindranath) [reply .github/workflows/nm-benchmark.yml:164]: Resolution : Lets keep it for now. We need to audit this permission at the company level. 
If this is not okay there is another option : 
   There is an option to point the action to a different repo - maybe we create one and have the action do all the write to that repo.

**Pull request merged**
